sfxclient/cancel14.go

```diff
@@ -12,14 +12,18 @@ type canceler interface {
 	CancelRequest(*http.Request)
 }
 
-func (h *HTTPSink) withCancel(ctx context.Context, req *http.Request) (err error) {
+func (h *HTTPSink) withCancel(ctx context.Context, req *http.Request, respValidator responseValidator) (err error) {
```

Comment:
we can get rid of cancel14.go and rename cancel15.go to cancel.go. We don't support either of those version and only the 1.5+ behavior. Also, it can probably just be moved into handleResponse or handleResponse moved into withCancel...  /shrug.